src/Compilers/Core/CodeAnalysisTest/Text/TextChangeTests.cs

```diff
@@ -880,6 +902,221 @@ public void TestMergeChanges_NoMiddleMan()
+                const int maxSkipLength = 2;
+                // generate sequence of "old edits" which meet invariants
+                for (int i = 0; i < originalText.Length; i += random.Next(maxSkipLength))
+                {
+                    var newText = string.Join("", Enumerable.Repeat('a', random.Next(oldMaxInsertLength)));
+                    var newChange = new TextChange(new TextSpan(i, length: random.Next(originalText.Length - i)), newText);
+                    i = newChange.Span.End;
+
+                    editedLength = editedLength - newChange.Span.Length + newChange.NewText.Length;
+                    oldChangesBuilder.Add(newChange);
+
+                    // Adjust as needed to generate a simpler reproducer for an error scenario
+                    if (oldChangesBuilder.Count == 5) break;
+                }
+
+                var change1 = originalText.WithChanges(oldChangesBuilder);
+
+                ArrayBuilder<TextChange> newChangesBuilder = ArrayBuilder<TextChange>.GetInstance();
+
+                // Adjust as needed to get a simpler error reproducer.
+                var newMaxInsertLength = editedLength * 2;
+                // generate sequence of "new edits" which meet invariants
+                for (int i = 0; i < editedLength; i += random.Next(maxSkipLength))
+                {
+                    var newText = string.Join("", Enumerable.Repeat('b', random.Next(newMaxInsertLength)));
+                    var newChange = new TextChange(new TextSpan(i, length: random.Next(editedLength - i)), newText);
+                    i = newChange.Span.End;
+
+                    newChangesBuilder.Add(newChange);
+
+                    // Adjust as needed to generate a simpler reproducer for an error scenario
+                    if (newChangesBuilder.Count == 5) break;
+                }
+
+                var change2 = change1.WithChanges(newChangesBuilder);
+                try
+                {
+                    var textChanges = change2.GetTextChanges(originalText);
+                    Assert.Equal(originalText.WithChanges(textChanges).ToString(), change2.ToString());
+                }
+                catch
+                {
+                    _output.WriteLine($@"
+    [Fact]
+    public void Fuzz_{iteration}()
+    {{
+        var originalText = SourceText.From(""{originalText}"");
+        var change1 = originalText.WithChanges({string.Join(", ", oldChangesBuilder.Select(c => c.GetDebuggerDisplay()))});
+        var change2 = change1.WithChanges({string.Join(", ", newChangesBuilder.Select(c => c.GetDebuggerDisplay()))});
+        Assert.Equal(""{change1}"", change1.ToString()); // double-check for correctness
+        Assert.Equal(""{change2}"", change2.ToString()); // double-check for correctness
+
+        var changes = change2.GetTextChanges(originalText);
+        Assert.Equal(""{change2}"", originalText.WithChanges(changes).ToString());
+    }}
+");
+                    throw;
```

Comment:
prior to your changes to Merge, did this generate any tests that hit an issue?

Reply:
See above. All of the `Fuzz_` tests came from this.